File: solutions/c/darts/1/darts.c
#include "darts.h"

uint8_t score(coordinate_t landing_position)
{
    float sum_of_squares = landing_position.x * landing_position.x + landing_position.y * landing_position.y;

    if (sum_of_squares <= 1)
        return 10;

    if (sum_of_squares <= 25)
        return 5;

    if (sum_of_squares <= 100)
        return 1;

    return 0;
}

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    final static double outerCircleRadius = 10;
    final static double middleCircleRadius = 5;
    final static double innerCircleRadius = 1;

    int score(double xOfDart, double yOfDart) {
        double distanceFromCenter = Math.sqrt(xOfDart * xOfDart + yOfDart * yOfDart);
        
        if (distanceFromCenter <= innerCircleRadius){
            return 10;
        } else if (distanceFromCenter <= middleCircleRadius) {
            return 5;
        } else if (distanceFromCenter <= outerCircleRadius) {
            return 1;
        }

        return 0;
    }
}
